### ISSUE TITLE
Codemod doesn't modify route templates with no controller

### ISSUE THREAD

**Issue by patocallaghan** [ember-codemods/ember-no-implicit-this-codemod#46]
Take the following app structure:

```
app/
  routes/
    -- application.js
  templates/
    -- application.hbs
```

The route returns a `model` of some sort and the template accesses `{{model}}`. There is no specific controller for the template.

Because the telemetry doesn't pick up that there is an associated controller for application.hbs it doesn't modify the file.

Is there a special mode we could add, `--route-template=true`?, which won't require the template to have a backing file to have it modified? This would allow us to modify `{{model}}` to be `{{this.model}}` (or `{{@model}}` from 3.14)

This won't just affect the `model` property but also others where we may use the `controller.set('someProperty', 'blah')` pattern in the route's `afterModel` hook.

**Comment by rwjblue** [comment]
I think in this case, when there is no controller file associated, we know:

* that these are route templates (assuming no `partial` use)
* that all non-helper / non-component named path expressions *must* reference the implicitly created controller context
* IFF the project is on ember-source@3.14+ we can use `@model` instead of `this.model`

**Comment by rwjblue** [comment]
tldr; I don't particularly think we need a new flag for this, I think we can infer what we need and do the "right thing" by default.

**Comment by patocallaghan** [comment]
@rwjblue I'm happy to put together a fix for this.

Is there any accurate way to determine the ember-source version?

**Comment by rwjblue** [comment]
`Ember.VERSION` 😸

**Comment by Turbo87** [comment]
@patocallaghan can you try again with `v1.0.0` or `master`? I think #62 might have fixed this issue.

**Comment by patocallaghan** [comment]
@Turbo87 Thanks! I'll give it a try 🙇

**Comment by efx** [comment]
@patocallaghan @Turbo87 I am seeing expected behavior so I think we can close this.

I used the below steps:

1. `ember new --no-welcome --yarn true my-app`
1. `cd my-app/`
1. `ember serve`
1. `npx ember-no-implicit-this-codemod http://localhost:4200 app/`

I added `{{model.name}}` to my `application.hbs` after step 2. After running the last step  it did show `{{this.model.name}}` as I expect.
